src/org/labkey/test/tests/CrossSiteScriptingForDeleteTest.java

```diff
@@ -0,0 +1,104 @@
+package org.labkey.test.tests;
+
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.labkey.test.BaseWebDriverTest;
+import org.labkey.test.Locator;
+import org.labkey.test.WebTestHelper;
+import org.labkey.test.categories.DailyB;
+import org.labkey.test.components.ext4.Window;
+import org.labkey.test.components.html.BootstrapMenu;
+import org.labkey.test.pages.reports.ManageViewsPage;
+
+import java.util.Collections;
+import java.util.List;
+
+@Category({DailyB.class})
+@BaseWebDriverTest.ClassTimeout(minutes = 3)
+public class CrossSiteScriptingForDeleteTest extends BaseWebDriverTest
+{
+    protected static final String PROJECT_NAME = "CrossSiteScriptingForDeleteTest";
+    protected static final String REPORT_NAME = "<span onclick='alert(\"fooled you!\")'>mouseover me, it's totally safe</span>";
+    private static final String LINK_REPORT_URL = "/project/home/begin.view";
+
+    @BeforeClass
+    public static void initTest()
+    {
+        CrossSiteScriptingForDeleteTest init = (CrossSiteScriptingForDeleteTest) getCurrentTest();
+        init.doSetup();
+    }
+
+    @Override
+    public List<String> getAssociatedModules()
+    {
+        return Collections.singletonList("core");
+    }
+
+    @Override
+    protected String getProjectName()
+    {
+        return PROJECT_NAME;
+    }
+
+    @Override
+    protected BrowserType bestBrowser()
+    {
+        return BrowserType.CHROME;
+    }
+
+    private void doSetup()
+    {
+        _containerHelper.createProject(PROJECT_NAME, null);
+
+    }
+
+    @Test
+    public void verifyReportNameIsEncoded()
+    {
+        goToProjectHome();
+        goToManageViews();
+
+        log("Adding the link report");
+        new BootstrapMenu.BootstrapMenuFinder(getDriver()).withButtonTextContaining("Add Report").find()
+                .clickSubMenu(true, "Link Report");
```

Comment:
Use the page returned by `goToManageViews`.
```suggestion
        log("Adding the link report");
        goToManageViews().clickAddReport("Link Report");
```

Reply:
Done